Purchases/Purchasing/StoreKitAbstractions/ProductType.swift

```diff
@@ -0,0 +1,90 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  ProductType.swift
+//
+//  Created by Nacho Soto on 2/15/22.
+
+import StoreKit
+
+extension StoreProduct {
+
+    /// The type of subscription of a product, whether a subscription or a one-time purchase.
+    ///
+    /// ### Related Symbols
+    /// - ``StoreProduct/ProductType-swift.enum``
+    @objc(RCStoreProductSubscriptionType)
+    public enum SubscriptionType: Int {
```

Comment:
i'm struggling w this name. i don't love `SubscriptionType` when one of the values is `subscription`... but i also don't have a better idea haha. thoughts:
HighLevelProductType
ProductTypeUmbrella
perhaps just doing isSubscription as a boolean

just spitballing, maybe there isn't a better way to do this.

Reply:
I forgot to mention that I wasn't sure about the naming and needed feedback, so these suggestions are very welcome!
I agree this is a bad name :/ Maybe `ProductCategory`?

Reply:
yeah i like that!

Reply:
Changed!